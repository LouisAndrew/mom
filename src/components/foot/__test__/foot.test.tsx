import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import Footer from '..';

describe('Footer component', () => {
    const el: React.ReactElement = <Footer />;
    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
