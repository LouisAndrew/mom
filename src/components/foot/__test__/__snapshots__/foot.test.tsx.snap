// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component matches snapshot 1`] = `
<footer
  className="sc-fzpjYC fzIDPV"
>
  <div
    className="sc-fzqBZW jJuffP"
    display="flex"
    height={
      Array [
        375,
        375,
        450,
      ]
    }
    overflow="hidden"
    width={
      Array [
        1,
        1,
        1,
        945,
      ]
    }
  >
    <div
      className="sc-fzqNJr cmOZrs"
    >
      <img
        alt="aesthetic interior"
        src="https://res.cloudinary.com/dsvdffre0/image/upload/v1596797190/jarek-ceborski-jn7uVeCdf6U-unsplash_n7xbxx.jpg"
      />
    </div>
    <div
      className="sc-fzoyAV flshlG"
      color="bg"
      display="flex"
      height="fit-content"
      width={
        Array [
          "80vw",
          "80vw",
          "560px",
        ]
      }
    >
      <h3
        className="sc-AxiKw dtruNh"
        fontFamily="heading"
        fontSize={
          Array [
            3,
            3,
            4,
            4,
          ]
        }
        width={
          Array [
            "fit-content",
            "fit-content",
            1,
          ]
        }
      >
        Tertarik untuk?
      </h3>
      <p
        className="sc-AxhUy hCxfQc"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        width={
          Array [
            "fit-content",
            "fit-content",
            1,
          ]
        }
      >
        Daftar ke mailing list kammiiii?
      </p>
      <label
        className="sc-fzokOt gsqCoe"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
          ]
        }
        fontWeight="bold"
        for="mailing-list"
        htmlFor="mailing-list"
        width={
          Array [
            1,
            1,
            0.5,
          ]
        }
      >
        Alamat email
        <input
          className="sc-AxheI hwZSkO"
          fontFamily="body"
          fontSize={
            Array [
              1,
              1,
              2,
              2,
            ]
          }
          fontWeight="bold"
          id="mailing-list"
          onChange={[Function]}
          placeholder="Isi alamat emailmu disini"
          width={
            Array [
              1,
              1,
              0.9,
            ]
          }
        />
      </label>
      <button
        className="sc-fznyAO gBqOlI"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        fontWeight="bold"
        height="fit-content"
        onClick={[Function]}
        width="fit-content"
      >
        Daftarkan email
      </button>
    </div>
  </div>
  <div
    className="sc-fznxsB hEXmCq"
    width={1}
  >
    <div
      className="sc-AxgMl cdtCDX"
      display="flex"
      width={1}
    >
      <svg
        fill="none"
        height="1em"
        id="logo-foot"
        viewBox="0 0 329 99"
        width="1em"
      >
        <path
          d="M65.272 74.24c-1.707 3.84-3.925 7.253-6.656 10.24a44.386 44.386 0 01-9.088 7.424 40.124 40.124 0 01-10.752 4.608 45.022 45.022 0 01-11.648 1.536c-8.363 0-14.848-2.475-19.456-7.424C3.064 85.589.76 77.611.76 66.688c0-3.84.299-8.15.896-12.928a99.296 99.296 0 013.2-14.592 90.597 90.597 0 015.632-14.464c2.304-4.693 5.12-8.875 8.448-12.544a39.345 39.345 0 0111.648-8.832C35.107 1.109 40.227 0 45.944 0c2.816 0 5.504.341 8.064 1.024 2.56.597 4.779 1.579 6.656 2.944 1.877 1.365 3.37 3.157 4.48 5.376 1.11 2.219 1.664 4.907 1.664 8.064 0 3.243-.683 5.973-2.048 8.192-1.365 2.219-3.67 3.328-6.912 3.328-1.024 0-2.219-.17-3.584-.512-1.28-.427-2.517-1.365-3.712-2.816 1.536-.768 2.901-2.304 4.096-4.608 1.195-2.39 1.792-4.736 1.792-7.04 0-2.133-.597-3.968-1.792-5.504-1.195-1.621-3.03-2.432-5.504-2.432-2.475 0-4.864.896-7.168 2.688-2.304 1.707-4.48 4.053-6.528 7.04-1.963 2.901-3.797 6.357-5.504 10.368-1.621 3.925-3.03 8.107-4.224 12.544a113.63 113.63 0 00-2.816 13.44c-.597 4.608-.896 9.045-.896 13.312 0 3.072.213 5.888.64 8.448.427 2.56 1.195 4.779 2.304 6.656 1.11 1.877 2.645 3.328 4.608 4.352 2.048 1.024 4.608 1.536 7.68 1.536 4.864 0 9.515-1.195 13.952-3.584a34.293 34.293 0 0011.136-9.856l2.944 1.28zm30.059-42.88c10.752 0 16.213 6.699 16.384 20.096.341.085.811.128 1.408.128 1.451 0 3.029-.213 4.736-.64a37.51 37.51 0 005.248-1.664 47.704 47.704 0 005.12-2.56 54.454 54.454 0 004.608-2.816l1.152 3.456c-2.475 2.645-5.717 4.907-9.728 6.784-3.925 1.792-8.192 3.072-12.8 3.84-.512 5.973-1.664 11.35-3.456 16.128-1.792 4.779-4.053 8.875-6.784 12.288-2.73 3.328-5.845 5.888-9.344 7.68a24.093 24.093 0 01-11.136 2.688c-2.987 0-5.76-.341-8.32-1.024-2.475-.768-4.608-1.963-6.4-3.584-1.792-1.621-3.2-3.712-4.224-6.272s-1.536-5.675-1.536-9.344c0-2.475.256-5.29.768-8.448.512-3.157 1.323-6.4 2.432-9.728a60.258 60.258 0 014.48-9.856 38.723 38.723 0 016.528-8.704 31.616 31.616 0 019.088-6.144c3.413-1.536 7.339-2.304 11.776-2.304zm-9.472 53.376c1.536 0 3.072-.64 4.608-1.92 1.621-1.365 3.072-3.2 4.352-5.504 1.365-2.39 2.56-5.205 3.584-8.448 1.024-3.243 1.792-6.741 2.304-10.496-1.365-.341-2.304-.981-2.816-1.92-.427-1.024-.64-2.176-.64-3.456 0-1.792.341-3.243 1.024-4.352.768-1.195 1.749-2.005 2.944-2.432-.171-2.645-.64-4.48-1.408-5.504-.683-1.11-1.835-1.664-3.456-1.664-2.304 0-4.48 1.28-6.528 3.84-2.048 2.475-3.84 5.547-5.376 9.216-1.536 3.584-2.773 7.424-3.712 11.52-.939 4.01-1.408 7.595-1.408 10.752 0 2.133.085 3.883.256 5.248.256 1.28.64 2.304 1.152 3.072a4.35 4.35 0 002.048 1.664c.853.256 1.877.384 3.072.384zm69.306-35.072c0-4.352-1.536-6.528-4.608-6.528-2.304 0-4.309 1.152-6.016 3.456-1.707 2.304-3.029 5.077-3.968 8.32L131.869 96h-18.432l13.568-64h18.432l-1.408 6.656c2.731-3.072 5.419-5.035 8.064-5.888 2.645-.853 4.949-1.28 6.912-1.28 4.096 0 7.467 1.152 10.112 3.456 2.731 2.219 4.096 5.888 4.096 11.008 0 2.56-.341 5.419-1.024 8.576a178.009 178.009 0 01-2.048 9.216c-.768 2.987-1.493 5.845-2.176 8.576-.597 2.645-.896 4.864-.896 6.656 0 1.621.341 2.944 1.024 3.968.768.939 2.133 1.408 4.096 1.408 1.365 0 2.56-.256 3.584-.768a9.28 9.28 0 002.816-2.304c.853-1.024 1.621-2.219 2.304-3.584a61.024 61.024 0 002.048-4.608h5.376c-1.792 5.29-3.712 9.515-5.76 12.672-2.048 3.072-4.139 5.419-6.272 7.04-2.133 1.621-4.224 2.688-6.272 3.2-1.963.512-3.797.768-5.504.768-2.816 0-5.205-.384-7.168-1.152-1.877-.853-3.413-1.963-4.608-3.328a12.285 12.285 0 01-2.56-4.608 19.488 19.488 0 01-.768-5.376c0-2.133.299-4.65.896-7.552a332.86 332.86 0 011.92-8.96 209.226 209.226 0 002.048-8.704c.597-2.901.896-5.376.896-7.424zm64.625 0c0-4.352-1.536-6.528-4.608-6.528-2.304 0-4.309 1.152-6.016 3.456-1.707 2.304-3.029 5.077-3.968 8.32L196.494 96h-18.432l13.568-64h18.432l-1.408 6.656c2.731-3.072 5.419-5.035 8.064-5.888 2.645-.853 4.949-1.28 6.912-1.28 4.096 0 7.467 1.152 10.112 3.456 2.731 2.219 4.096 5.888 4.096 11.008 0 2.56-.341 5.419-1.024 8.576a178.009 178.009 0 01-2.048 9.216c-.768 2.987-1.493 5.845-2.176 8.576-.597 2.645-.896 4.864-.896 6.656 0 1.621.341 2.944 1.024 3.968.768.939 2.133 1.408 4.096 1.408 1.365 0 2.56-.256 3.584-.768a9.28 9.28 0 002.816-2.304c.853-1.024 1.621-2.219 2.304-3.584a61.024 61.024 0 002.048-4.608h5.376c-1.792 5.29-3.712 9.515-5.76 12.672-2.048 3.072-4.139 5.419-6.272 7.04-2.133 1.621-4.224 2.688-6.272 3.2-1.963.512-3.797.768-5.504.768-2.816 0-5.205-.384-7.168-1.152-1.877-.853-3.413-1.963-4.608-3.328a12.285 12.285 0 01-2.56-4.608 19.488 19.488 0 01-.768-5.376c0-2.133.299-4.65.896-7.552a332.86 332.86 0 011.92-8.96 209.226 209.226 0 002.048-8.704c.597-2.901.896-5.376.896-7.424zm39.025-34.688c0-2.816.981-5.205 2.944-7.168 1.963-2.048 4.352-3.072 7.168-3.072s5.205 1.024 7.168 3.072c2.048 1.963 3.072 4.352 3.072 7.168s-1.024 5.205-3.072 7.168c-1.963 1.963-4.352 2.944-7.168 2.944s-5.205-.981-7.168-2.944-2.944-4.352-2.944-7.168zM274.687 32l-9.216 43.52c-.341 1.45-.512 2.816-.512 4.096 0 1.792.427 3.03 1.28 3.712.853.683 2.176 1.024 3.968 1.024 2.475 0 4.779-1.067 6.912-3.2s3.669-4.821 4.608-8.064h5.376c-1.792 5.205-3.883 9.387-6.272 12.544-2.389 3.072-4.821 5.461-7.296 7.168-2.475 1.621-4.907 2.688-7.296 3.2-2.304.512-4.309.768-6.016.768-3.84 0-7.083-1.195-9.728-3.584-2.645-2.475-3.968-6.4-3.968-11.776 0-2.304.341-5.077 1.024-8.32L256.255 32h18.432zm24.397 38.144l-.256 2.048v1.92c0 1.28.085 2.56.256 3.84a11.236 11.236 0 001.28 3.328 6.645 6.645 0 002.432 2.432c1.024.597 2.347.896 3.968.896 2.475 0 5.077-.683 7.808-2.048 2.816-1.365 5.248-2.944 7.296-4.736-.683 3.67-1.749 6.741-3.2 9.216-1.365 2.39-3.029 4.31-4.992 5.76a18.374 18.374 0 01-6.528 3.072 30.495 30.495 0 01-7.424.896c-5.973 0-10.795-1.579-14.464-4.736-3.584-3.157-5.376-8.363-5.376-15.616 0-2.475.256-5.29.768-8.448.512-3.157 1.28-6.4 2.304-9.728a54.742 54.742 0 014.352-9.728 38.723 38.723 0 016.528-8.704c2.56-2.56 5.547-4.608 8.96-6.144 3.413-1.536 7.296-2.304 11.648-2.304 2.816 0 5.12.341 6.912 1.024 1.877.683 3.328 1.621 4.352 2.816 1.109 1.11 1.835 2.39 2.176 3.84.427 1.45.64 2.987.64 4.608 0 3.67-.811 7.083-2.432 10.24-1.621 3.072-3.797 5.803-6.528 8.192-2.731 2.304-5.888 4.181-9.472 5.632a36.839 36.839 0 01-11.008 2.432zm19.584-26.752c0-1.621-.256-2.901-.768-3.84-.427-.939-1.237-1.408-2.432-1.408-1.621 0-3.2.81-4.736 2.432-1.536 1.536-2.987 3.541-4.352 6.016-1.365 2.475-2.603 5.248-3.712 8.32A77.955 77.955 0 00299.98 64c2.475-.085 4.821-.725 7.04-1.92a21.322 21.322 0 006.016-4.608 22.767 22.767 0 004.096-6.4 19.901 19.901 0 001.536-7.68z"
          fill="#2E282A"
        />
      </svg>
      <ul
        className="sc-fzoLag feeFZQ"
      >
        <li
          className="sc-fzoXzr kOcIvj"
          fontFamily="heading"
          fontSize={
            Array [
              2,
              2,
            ]
          }
        >
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="sc-fzoXzr kOcIvj"
          fontFamily="heading"
          fontSize={
            Array [
              2,
              2,
            ]
          }
        >
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Katalog
          </a>
        </li>
      </ul>
      <div
        className="sc-fznJRM gMzkig"
        display="flex"
      >
        <span
          className="sc-fznWqX embqfE"
          color="bg"
          fontFamily="heading"
          fontSize={
            Array [
              1,
              1,
              2,
            ]
          }
        >
          <svg
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path d=\\"M8 11a1 1 0 1 0 1 1a1 1 0 0 0-1-1zm4 0a1 1 0 1 0 1 1a1 1 0 0 0-1-1zm4 0a1 1 0 1 0 1 1a1 1 0 0 0-1-1zm-4-9A10 10 0 0 0 2 12a9.89 9.89 0 0 0 2.26 6.33l-2 2a1 1 0 0 0-.21 1.09A1 1 0 0 0 3 22h9a10 10 0 0 0 0-20zm0 18H5.41l.93-.93a1 1 0 0 0 .3-.71a1 1 0 0 0-.3-.7A8 8 0 1 1 12 20z\\" fill=\\"#F3F0F0\\"/>",
              }
            }
            focusable={false}
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "transform": "rotate(360deg)",
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          />
          +6281808273193
        </span>
        <span
          className="sc-fznWqX embqfE"
          color="bg"
          fontFamily="heading"
          fontSize={
            Array [
              1,
              1,
              2,
            ]
          }
        >
          <svg
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path d=\\"M19 4H5a3 3 0 0 0-3 3v10a3 3 0 0 0 3 3h14a3 3 0 0 0 3-3V7a3 3 0 0 0-3-3zm-.41 2l-5.88 5.88a1 1 0 0 1-1.42 0L5.41 6zM20 17a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V7.41l5.88 5.88a3 3 0 0 0 4.24 0L20 7.41z\\" fill=\\"#F3F0F0\\"/>",
              }
            }
            focusable={false}
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "transform": "rotate(360deg)",
                "verticalAlign": "-0.125em",
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          />
          mock-email@yahoo.dev
        </span>
      </div>
    </div>
  </div>
</footer>
`;
